### ISSUE TITLE
[bug] SSR hydration error with Suspense and OS dark mode color preference

### ISSUE THREAD

**Issue by nderscore** [tamagui/tamagui#504]
**Describe the bug**

When a Next.js application using tamagui (with the recommended theme swapping method) renders a suspense boundary containing Tamagui components, and the user has dark mode color preference set `prefers-color-scheme: dark`, the following error is thrown:

> This Suspense boundary received an update before it finished hydrating. This caused the boundary to switch to client rendering. The usual way to fix this is to wrap the original update in startTransition.

**To Reproduce**

See bug reproduction repo here: https://github.com/nderscore/tamagui-suspense-demo

Changes from the starter:

* Nest a [very-bare-bones suspense boundary](https://github.com/nderscore/tamagui-suspense-demo/blob/main/packages/app/components/SuspenseBoundary.tsx) component inside global providers:
  * https://github.com/nderscore/tamagui-suspense-demo/blob/main/packages/app/provider/index.tsx#L10
* Add timer-delayed mock suspense API call to user details page:
  * https://github.com/nderscore/tamagui-suspense-demo/blob/main/packages/app/features/user/detail-screen.tsx#L24

**Comment by natew** [comment]
Did you try adding the `startTransition` around that?

**Comment by natew** [comment]
Accidental close

**Comment by nderscore** [comment]
> Did you try adding the `startTransition` around that?

Yep! that was the fix. 

I submitted a PR to update the starter examples in #506

**Comment by natew** [comment]
Thank you

**Comment by nderscore** [comment]
@natew just a heads up: I started seeing this same hydration error happening again starting on v1.45.3 and above. v1.45.2 still works on my project.

It looks like the latest starter template [no longer has `startTransition` wrapping the call to setTheme](https://github.com/tamagui/tamagui/blob/master/starters/next-expo-solito/apps/next/pages/_app.tsx#L37), but removing that from my `_app.tsx` doesn't seem to make any difference to the warning.

Will investigate this more fully soon and make a reproduction. It's possible there's some other way I've deviated from the starter's next.js app too 🤔
